worlds/v6/Rules.py

```diff
@@ -1,43 +1,44 @@
-    area_cost_map.update({0: 0})
-
-    for i in range(1, 5):
-        connect_regions(world, player, "Menu", v6areas[area_connections[i] - 1],
-                        lambda state, i=i: _has_trinket_range(state, player,
-                                                              world.DoorCost[player].value * (area_cost_map[i] - 1),
-                                                              world.DoorCost[player].value * area_cost_map[i]))
+        return state.has(f"Trinket {str(i + 1).zfill(2)}", player)
+
+
+def can_reach_all_areas(state: CollectionState, player: int) -> bool:
+    return state.can_reach("Laboratory", "Entrance", player) and \
+           state.can_reach("The Tower", "Entrance", player) and \
+           state.can_reach("Space Station 2", "Entrance", player) and \
+           state.can_reach("Warp Zone", "Entrance", player)
+
+
+def can_reach_npc_trinket(state: CollectionState, player: int) -> bool:
+    return state.can_reach("Laboratory", "Entrance", player) or (
+           state.can_reach("The Tower", "Entrance", player) and
+           state.can_reach("Space Station 2", "Entrance", player) and
+           state.can_reach("Warp Zone", "Entrance", player))
+
+
+def set_rules(multiworld: MultiWorld, player: int):
+    for index, region in enumerate(v6_areas, start=1):
+        multiworld.get_entrance(region, player).access_rule = lambda state: has_trinket_range(
+            state,
+            player,
+            multiworld.DoorCost[player] * (multiworld.worlds[player].area_cost_map[index] - 1),
+            multiworld.DoorCost[player] * multiworld.worlds[player].area_cost_map[index],
+        )
 
     # Special Rule for V
-    add_rule(world.get_location("V", player), lambda state: state.can_reach("Laboratory", 'Region', player) and
-                                                            state.can_reach("The Tower", 'Region', player) and
-                                                            state.can_reach("Space Station 2", 'Region', player) and
-                                                            state.can_reach("Warp Zone", 'Region', player))
+    add_rule(multiworld.get_location("V", player), lambda state: can_reach_all_areas(state, player))
 
     # Special Rule for NPC Trinket
-    add_rule(world.get_location("NPC Trinket", player),
-             lambda state: state.can_reach("Laboratory", 'Region', player) or
-                           (state.can_reach("The Tower", 'Region', player) and
-                            state.can_reach("Space Station 2", 'Region', player) and
-                            state.can_reach("Warp Zone", 'Region', player)))
+    add_rule(multiworld.get_location("NPC Trinket", player), lambda state: can_reach_npc_trinket(state, player))
 
-    world.completion_condition[player] = lambda state: state.can_reach("V", 'Location', player)
+    # Victory condition
+    if multiworld.DoorCost[player] == 0:
```

Comment:
Please just use the original completion condition, no need to differentiate.

Reply:
Helps if my logic above was correct. Fixed.